lib/cocoapods/installer.rb

```diff
@@ -502,26 +504,37 @@ def verify_no_lockfile_changes!
     # @return [void]
     #
     def install_pod_sources
+      @downloaded_specs = []
       @installed_specs = []
       pods_to_install = sandbox_state.added | sandbox_state.changed
       title_options = { :verbose_prefix => '-> '.green }
-      root_specs.sort_by(&:name).each do |spec|
-        if pods_to_install.include?(spec.name)
-          if sandbox_state.changed.include?(spec.name) && sandbox.manifest
-            current_version = spec.version
-            previous_version = sandbox.manifest.version(spec.name)
-            has_changed_version = current_version != previous_version
-            current_repo = analysis_result.specs_by_source.detect { |key, values| break key if values.map(&:name).include?(spec.name) }
-            current_repo &&= (Pod::TrunkSource::TRUNK_REPO_NAME if current_repo.name == Pod::TrunkSource::TRUNK_REPO_NAME) || current_repo.url || current_repo.name
-            previous_spec_repo = sandbox.manifest.spec_repo(spec.name)
-            has_changed_repo = !previous_spec_repo.nil? && current_repo && !current_repo.casecmp(previous_spec_repo).zero?
-            title = "Installing #{spec.name} #{spec.version}"
-            title << " (was #{previous_version} and source changed to `#{current_repo}` from `#{previous_spec_repo}`)" if has_changed_version && has_changed_repo
-            title << " (was #{previous_version})" if has_changed_version && !has_changed_repo
-            title << " (source changed to `#{current_repo}` from `#{previous_spec_repo}`)" if !has_changed_version && has_changed_repo
-          else
-            title = "Installing #{spec}"
+
+      sorted_root_specs = root_specs.sort_by(&:name)
+
+      # Download pods in parallel before installing if the option is set
```

Comment:
That's true, but I wonder if it would be safer to leave the parallel option in place to act as a feature flag of sorts for the new downloading logic, just in case something goes wrong

Reply:
It is true that semantically we will introduce another thread even if its just 1 versus the current implementation. I am OK to proceed as such.